src/umpire/strategy/FixedPool.cpp

```diff
@@ -194,6 +194,17 @@ void FixedPool::deallocate(void* ptr)
   UMPIRE_ERROR("Could not find the pointer to deallocate");
 }
 
+void FixedPool::release() 
+{ 
+  for (auto& p : m_pool) {
+    if (m_obj_per_pool == p.num_avail) {
+      p.strategy->deallocate(p.data);
+      std::free(p.avail);
+    } 
+  }
+  m_pool.clear(); 
```

Comment:
This shouldn't be `clear`, since you only want to remove the elements where `p.num_avail == m_obj_per_pool`. You could use `remove_if` to do this.

Reply:
Ah, makes sense. Thanks